src/uu/cp/src/platform/linux.rs

```diff
@@ -132,6 +136,18 @@ where
     Ok(num_bytes_copied)
 }
 
+fn check_sparse_detection(source: &Path, sparse_flag: &mut bool) -> std::io::Result<()> {
+    let src_file = File::open(source)?;
+
+    use std::os::unix::prelude::MetadataExt;
+
+    let size: usize = src_file.metadata()?.size().try_into().unwrap();
```

Comment:
what happens if the unwrap fails ? (same question for the line below)

Reply:
The unwrap shouldn't fail as both blocks and size are u64 .

Reply:
But `usize` might be 32-bit right?

Reply:
> But `usize` might be 32-bit right?
Yes , the usize might 32-bit . That is very likely the reason for my builds failing . I will look into some ways to doing a safe conversion for my purposes.